ui.frontend/src/main/webpack/site/js/components/searchResults.ts

```diff
@@ -1,24 +1,28 @@
-import { OnSearchItemClickCallback } from '../types/callbacks'
+import { createCustomEvent, events } from '../service/serviceEvent'
 import { buildSearchItem, SearchItem } from './searchItem'
 
 type SearchResultsOptions = {
   searchItems: SearchItem[]
   tabId: string
-  onSearchItemClick?: OnSearchItemClickCallback
 }
 
 export const generateSearchItemList = (
   searchItems: SearchItem[],
-  onSearchItemClick?: OnSearchItemClickCallback,
 ): HTMLDivElement[] => {
   return searchItems.map((searchItem) => {
     const searchItemElement = buildSearchItem(searchItem)
-
-    if (onSearchItemClick) {
-      searchItemElement.addEventListener('click', (event) => {
-        event.preventDefault()
-
-        onSearchItemClick?.(searchItem.url)
+    if (searchItemElement) {
```

Comment:
**Thought**

searchItemElement was null before? I see you added a check for it.
If it can be null you can have searchItemElement?.addEventListener though.

Reply:
I removed the check for it and wrote it as you suggested it